pkg/operator/controller/ingress/controller.go

```diff
@@ -83,10 +85,12 @@ var (
 // in the manager namespace.
 func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	reconciler := &reconciler{
-		config:   config,
-		client:   mgr.GetClient(),
-		cache:    mgr.GetCache(),
-		recorder: mgr.GetEventRecorderFor(controllerName),
+		config:        config,
+		client:        mgr.GetClient(),
+		cache:         mgr.GetCache(),
+		recorder:      mgr.GetEventRecorderFor(controllerName),
+		routeMap:      make(map[types.NamespacedName]routev1.TLSTerminationType),
```

Comment:
The `routeMap` and `shardRouteMap` should not be added to the reconciler and updated every time a `Route` is observed by the watcher.  The maps should be in the cache.